splunk/tests/splunk_test_prep.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// BuildSplunk builds the Splunk image used for testing Splunk queries.
+func BuildSplunk() {
+	buildCmd := exec.Command("podman", "build", "-t", "test_splunk", "./splunk")
+
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+	err := buildCmd.Run()
+	if err != nil {
+		log.Fatalf("Failed to build image: %v", err)
+	}
+	log.Println("Image build done successfully!")
+}
+
+// RunSplunk deploy the Splunk container used for testing Splunk queries.
+func RunSplunk() {
+	filePath := "runtime.txt"
+	file, err := os.Create(filePath)
+	if err != nil {
+		log.Fatalf("Failed to create file: %v", err)
+	}
+	defer file.Close()
+
+	cmd := exec.Command("podman", "run", "-e", "SPLUNK_START_ARGS=--accept-license", "-e", "SPLUNK_PASSWORD=Password", "-p", "8000:8000", "-p", "8089:8089", "localhost/test_splunk:latest")
+	cmd.Stdout = file
```

Comment:
I don't think we need to route the command output to a file like that - we can see the container output with `podman logs` if we need to.

We can let any stray lines go the the usual stdout I think, especially if we'll use the `-d` option for `podman run`. The default handling of stderr by `cmd.Run` or `cmd.Output` seems sufficient to me.

Reply:
Ok, I see you are using the file as a monitoring device.... Why use a file when you can use a simple descriptor instead? 

The problem with this approach is that the podman process remains attached to our Go process and we must make sure to maintain it and clean up after it. It would probably be better to use `-d` and monitor the service network port as a way of knowing if its up - monitoring the port would also be what we need when we run in a cluster.

Reply:
Monitoring the port isn't enough, it's the same issue we had with the shell script in the first stage of the dockerfile... the port & the availability endpoints are up before the service is completely up and ready

Reply:
Still, we should find a network - based approach. The log would not be available when running in a cluster. Using it locally can create all kinds of complications too.

Reply:
I implemented the approach from the first comment for now, while I search for a network-based approach. 
but I'm not sure what can we use that we didn't already try in the dockerfile. 
maybe we can loop for 2 min trying query splunk for `eventcount index=test_index` and see that we have *something*  in, but it feels kinda backwards

Reply:
how do we know the service is not ready when the availability endpoint says it does?

Reply:
I tested it before, where a script checked for the availability endpoint (which responded positively before the playbook finished), and proceeded to do data-indexing only to get an error.

Reply:
well perhaps it will behave differently now that we're not indexing data but reading it

Reply:
It seems like we can use `https://localhost:8089/services/server/status`
while it returns 200 a few steps before the ansible playbook finish running, , running a query at the same time/stage returns the expected result.

Reply:
I thought that would be the case, the availability signal belongs to the querying front-end not the back-end infestation engine